nmagent/client_test.go

```diff
@@ -432,3 +432,73 @@ func TestNMAgentDeleteNC(t *testing.T) {
 		})
 	}
 }
+
+func TestNMAgentSupportedAPIs(t *testing.T) {
+	tests := []struct {
+		name      string
+		exp       nmagent.HomeAzInfo
+		expPath   string
+		resp      map[string]interface{}
+		shouldErr bool
+	}{
+		{
+			"happy path",
+			nmagent.HomeAzInfo{HomeAz: "az01"},
+			"/machine/plugins/?comp=nmagent&type=GetHomeAzInfo",
+			map[string]interface{}{
+				"httpStatusCode": "200",
+				"HomeAz":         "az01",
+				"DcmtRegion":     "uswest2",
+			},
+			false,
+		},
+		{
+			"empty response",
+			nmagent.HomeAzInfo{},
+			"/machine/plugins/?comp=nmagent&type=GetHomeAzInfo",
+			map[string]interface{}{
+				"httpStatusCode": "404",
```

Comment:
they should never return 404 if the api is supported. It can be 500. It can also be 401 - in which case we need to treat it as fatal / non-retryable error. We should create alerts on those